src/main/java/teammates/logic/core/FeedbackQuestionsLogic.java

```diff
@@ -421,6 +473,48 @@ public int compare(FeedbackQuestionAttributes q1, FeedbackQuestionAttributes q2)
                 recipients.put(student.email, student.name);
             }
             break;
+        case CUSTOM:
+            List<StudentAttributes> studentList = studentsLogic.getStudentsForCourse(question.courseId);
+            Map<String, String> studentEmailToStudentNameMap = new HashMap<String, String>();
+            List<InstructorAttributes> instructorList =
+                    instructorsLogic.getInstructorsForCourse(question.courseId);
+            Map<String, String> instructorEmailToInstructorNameMap = new HashMap<String, String>();
+            
+            for (StudentAttributes student : studentList) {
+                studentEmailToStudentNameMap.put(student.getEmail(), student.getName());
+            }
+            for (InstructorAttributes instructor : instructorList) {
+                instructorEmailToInstructorNameMap.put(instructor.getEmail(), instructor.getName());
+            }
+            
+            for (FeedbackPathAttributes feedbackPath : question.feedbackPaths) {
+                boolean isUserStudentAndFeedbackPathGiver =
+                        isStudentGiver
+                        && feedbackPath.isStudentFeedbackPathGiver(studentGiver);
+                
+                boolean isUserInstructorAndFeedbackPathGiver =
+                        isInstructorGiver
+                        && feedbackPath.isInstructorFeedbackPathGiver(instructorGiver.getEmail());
+                
+                boolean isUserFeedbackPathGiver =
+                        isUserStudentAndFeedbackPathGiver || isUserInstructorAndFeedbackPathGiver;
+                
+                if (isUserFeedbackPathGiver) {
+                    String feedbackPathRecipientId = feedbackPath.getRecipientId();
+                    String name = "";
+                    
+                    if (studentEmailToStudentNameMap.containsKey(feedbackPathRecipientId)) {
+                        name = studentEmailToStudentNameMap.get(feedbackPathRecipientId);
+                    } else if (instructorEmailToInstructorNameMap.containsKey(feedbackPathRecipientId)) {
+                        name = instructorEmailToInstructorNameMap.get(feedbackPathRecipientId);
+                    } else {
+                        name = feedbackPathRecipientId;
+                    }
+                    
+                    recipients.put(feedbackPathRecipientId, name);
+                }
+            }
+            break;
```

Comment:
It's not very SLAP to have long code blocks inside case statements. Abstract into methods.